data/test/macros/wml_unit_test_macros.cfg

```diff
@@ -116,29 +125,46 @@ bob#endarg
         max_hitpoints = 1000
     [/modify_unit]
 
-    [test_do_attack_by_id]
-        attacker={ID1}
-        defender={ID2}
-    [/test_do_attack_by_id]
-
-    [store_unit]
-        [filter]
-            id={ID1}
-        [/filter]
-        variable={ID1}
-    [/store_unit]
-    [store_unit]
-        [filter]
-            id={ID2}
-        [/filter]
-        variable={ID2}
-    [/store_unit]
-
-    {ASSERT {VARIABLE_CONDITIONAL {ID1}.hitpoints numerical_equals "$(${ID1}.max_hitpoints-{DAMAGE1})"}}
-    {ASSERT {VARIABLE_CONDITIONAL {ID2}.hitpoints numerical_equals "$(${ID2}.max_hitpoints-{DAMAGE2})"}}
-
-    {CLEAR_VARIABLE {ID1}}
-    {CLEAR_VARIABLE {ID2}}
+	{VARIABLE weapons {WEAPON_COUNT}}
+	[for]
+		start=0
+		end="$($weapons-1)"
+		step=1
+		[do]
+			[test_do_attack_by_id]
+				attacker={ID1}
+				defender={ID2}
+				weapon=$i
+                resupply_attacks_left=1
+			[/test_do_attack_by_id]
+
+            [store_unit]
+                [filter]
+                    id={ID1}
+                [/filter]
+                variable={ID1}
+            [/store_unit]
+            [store_unit]
+                [filter]
+                    id={ID2}
+                [/filter]
+                variable={ID2}
+            [/store_unit]
+
+            {ASSERT {VARIABLE_CONDITIONAL {ID1}.hitpoints numerical_equals "$(${ID1}.max_hitpoints-{DAMAGE1})"}}
+            {ASSERT {VARIABLE_CONDITIONAL {ID2}.hitpoints numerical_equals "$(${ID2}.max_hitpoints-{DAMAGE2})"}}
+
+            [modify_unit]
+                [filter]
+                    id={ID1},{ID2}
+                [/filter]
+                hitpoints = 1000
```

Comment:
Doing this at the start of the loop would seem more logical to me, and that should make it easier to examine the state if the test fails too.

Reply:
Updated.